Problems with new version of Flask 0.11

### Description

New version [0.11 of Flask](http://flask.pocoo.org/docs/0.11/changelog/#version-0-11) change some behaviour that Connexion currently is not supporting.
### Expected bahaviour

No errors while running the tests.

### Actual behaviour

We get this exception:

```
.tox/py35/lib/python3.5/site-packages/werkzeug/test.py:778: in get
    return self.open(*args, **kw)
.tox/py35/lib/python3.5/site-packages/flask/testing.py:113: in open
    follow_redirects=follow_redirects)
.tox/py35/lib/python3.5/site-packages/werkzeug/test.py:751: in open
    response = self.run_wsgi_app(environ, buffered=buffered)
.tox/py35/lib/python3.5/site-packages/werkzeug/test.py:668: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
.tox/py35/lib/python3.5/site-packages/werkzeug/test.py:871: in run_wsgi_app
    app_rv = app(environ, start_response)
.tox/py35/lib/python3.5/site-packages/flask/app.py:2000: in __call__
    return self.wsgi_app(environ, start_response)
.tox/py35/lib/python3.5/site-packages/flask/app.py:1991: in wsgi_app
    response = self.make_response(self.handle_exception(e))
.tox/py35/lib/python3.5/site-packages/flask/app.py:1559: in handle_exception
    handler = self._find_error_handler(InternalServerError())
.tox/py35/lib/python3.5/site-packages/flask/app.py:1481: in _find_error_handler
    return find_handler(self.error_handler_spec[None].get(code))
----------------------------------------------------------------------------------------------------------------
handler_map = <function App.common_error_handler at 0x102aa67b8>

    def find_handler(handler_map):
        if not handler_map:
            return
        queue = deque(exc_class.__mro__)
        # Protect from geniuses who might create circular references in
        # __mro__
        done = set()

        while queue:
            cls = queue.popleft()
            if cls in done:
                continue
            done.add(cls)
>           handler = handler_map.get(cls)
E           AttributeError: 'function' object has no attribute 'get'

.tox/py35/lib/python3.5/site-packages/flask/app.py:1465: AttributeError

```

### Steps to reproduce

Run tests:

```
$ tox -r -e py35
```

### Additional info:

Output of the commands:

- `python --version`: 3.5.1
- `pip show connexion | grep "^Version\:"`: 1.0.97

--- a/connexion/app.py
+++ b/connexion/app.py
@@ -71,7 +71,7 @@ class App(object):
         logger.debug('Specification directory: %s', self.specification_dir)
 
         logger.debug('Setting error handlers')
-        for error_code in range(400, 600):  # All http status from 400 to 599 are errors
+        for error_code in werkzeug.exceptions.default_exceptions:
             self.add_error_handler(error_code, self.common_error_handler)
 
         self.port = port
@@ -153,7 +153,7 @@ class App(object):
         :type error_code: int
         :type function: types.FunctionType
         """
-        self.app.error_handler_spec[None][error_code] = function
+        self.app.register_error_handler(error_code, function)
 
     def add_url_rule(self, rule, endpoint=None, view_func=None, **options):
         """

--- a/tests/api/test_errors.py
+++ b/tests/api/test_errors.py
@@ -62,7 +62,7 @@ def test_errors(problem_app):
     assert problematic_json.status_code == 500
 
     custom_problem = app_client.get('/v1.0/customized_problem_response')
-    assert custom_problem.status_code == 402
+    assert custom_problem.status_code == 403
     problem_body = json.loads(custom_problem.data.decode('utf-8'))
     assert 'amount' in problem_body
     assert problem_body['amount'] == 23.

--- a/tests/fakeapi/hello.py
+++ b/tests/fakeapi/hello.py
@@ -335,7 +335,7 @@ def get_invalid_response():
 
 
 def get_custom_problem_response():
-    return problem(402, "You need to pay", "Missing amount",
+    return problem(403, "You need to pay", "Missing amount",
                    ext={'amount': 23.0})
 
 

--- a/tox.ini
+++ b/tox.ini
@@ -3,7 +3,7 @@ max-line-length=120
 exclude=connexion/__init__.py
 
 [tox]
-envlist=pypy,py27,py34,py35,isort-check,flake8
+envlist=pypy,py27,py34,py35,py35-flask0.10.1,isort-check,flake8
 
 [tox:travis]
 pypy=pypy
@@ -22,3 +22,6 @@ commands=python setup.py flake8
 basepython=python3
 deps=isort
 commands=isort -ns __init__.py -rc -c {toxinidir}/connexion {toxinidir}/examples {toxinidir}/tests
+
+[testenv:py35-flask0.10.1]
+deps=flask==0.10.1

